### PR TITLE
release: bump to version 1.2.0rc1

### PR DESCRIPTION
**Pull request by finswimmer**
_Requires release of poetry-core 1.1.0rc1 first https://github.com/python-poetry/poetry-core/pull/443_

### Added

- Added support for subdirectories in git dependencies ([#5172](https://github.com/python-poetry/poetry/pull/5172))
- Added support for yanked releases and files (PEP-592) ([#5841](https://github.com/python-poetry/poetry/pull/5841))
- Virtual environments can now be created even with empty project names ([#5856](https://github.com/python-poetry/poetry/pull/5856))
- Added support for `nushell` in `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))

### Changed

- Poetry now falls back to gather metadata for dependencies via pep517 if parsing `pyproject.toml` fails ([#5834](https://github.com/python-poetry/poetry/pull/5834))
- Replaced Poetry's helper method `canonicalize_name()` with `packaging.utils.canonicalize_name()` ([#6022](https://github.com/python-poetry/poetry/pull/6022))
- Removed code for the `export` command, which is now provided via plugin ([#6128](https://github.com/python-poetry/poetry/pull/6128))
- Extras and extras dependencies are now sorted in the lock file ([#6169](https://github.com/python-poetry/poetry/pull/6169))
- Removed deprecated (1.2-only) CLI options ([#6210](https://github.com/python-poetry/poetry/pull/6210))

### Fixed

- Fixed an issue where symlinks in the lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))
- Fixed a `tomlkit` regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))
- Fixed an issue where the `POETRY_PYPI_TOKEN_PYPI` environment variable wasn't respected ([#5911](https://github.com/python-poetry/poetry/pull/5911))
- Fixed an issue where neither Python nor a managed venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))
- Improved error message of `poetry publish` in the event of an upload error ([#6043](https://github.com/python-poetry/poetry/pull/6043))
- Fixed an issue where `poetry lock` fails without output ([#6058](https://github.com/python-poetry/poetry/pull/6058))
- Fixed an issue where Windows drive mappings break virtual environment names ([#6110](https://github.com/python-poetry/poetry/pull/6110))
- `tomlkit` versions with memory leak are now avoided ([#6160](https://github.com/python-poetry/poetry/pull/6160))
- Fixed an infinite loop in the solver ([#6178](https://github.com/python-poetry/poetry/pull/6178))
- Fixed an issue where latest version was used instead of locked one for vcs dependencies with extras ([#6185](https://github.com/python-poetry/poetry/pull/6185))

### Docs

- Document use of the `subdirectory` parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
- Document suggested `tox` config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))

### CODE REVIEW

**Comment by radoering** [comment]
Bumping to the next poetry-core release requires #6163. I added a vcs dev dependency on poetry-core in this PR so we can also merge it and remove the dev dependency when bumping poetry-core. Alternatively, the PR can be updated as soon as a core release is done.

**Comment by finswimmer** [comment]
>  Alternatively, the PR can be updated as soon as a core release is done.

IMO this is the more clean way and we should do it like this.

**Comment by dimbleby** [comment]
There are a handful of fixes in cleo since 1.0.0a5 which fix bugs reported in this repository.

I'd much rather that poetry just got on and cut its own release than that it get blocked on cleo: but if a cleo release is just a button-press away then it would be nice to do that and pick it up here.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # Change Log
 
+## [1.2.0rc1] - 2022-08-16
+
+### Added
+- Virtual environments can now be created even with empty project name ([#5856](https://github.com/python-poetry/poetry/pull/5856))
+- Added support for subdirectories in git dependencies ([#5172](https://github.com/python-poetry/poetry/pull/5172))
+- Added support for nushell on `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))
```
Comment (mkniewallner) [inline CHANGELOG.md:8]:
```suggestion
- Added support for `nushell` on `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+
+
+### Changed
+- extras and extras dependencies are now sorted in lockfile ([#6169](https://github.com/python-poetry/poetry/pull/6169))
```
Comment (mkniewallner) [inline CHANGELOG.md:12]:
```suggestion
- Extras and extras dependencies are now sorted in lock file ([#6169](https://github.com/python-poetry/poetry/pull/6169))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+- Removed code for the `export` command, which is now provided via plugin ([#6128](https://github.com/python-poetry/poetry/pull/6128))
+- Replaced Poetry's helper method canonicalize_name() by packaging.utils.canonicalize_name() ([#6022](https://github.com/python-poetry/poetry/pull/6022))
```
Comment (mkniewallner) [inline CHANGELOG.md:14]:
```suggestion
- Replaced Poetry's helper method `canonicalize_name()` by `packaging.utils.canonicalize_name()` ([#6022](https://github.com/python-poetry/poetry/pull/6022))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,33 @@
...
+
+### Fixed
+- tomlkit versions with memory leak are now avoided ([#6160](https://github.com/python-poetry/poetry/pull/6160))
```
Comment (mkniewallner) [inline CHANGELOG.md:17]:
```suggestion
- `tomlkit` versions with memory leak are now avoided ([#6160](https://github.com/python-poetry/poetry/pull/6160))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,33 @@
...
+- Fixed an issue where Windows drive mappings break virtual environment names ([#6110](https://github.com/python-poetry/poetry/pull/6110))
+- Fixed an issue where symlinks in lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))
+- Fixed an issue where `poetry lock` fails without output ([#6058](https://github.com/python-poetry/poetry/pull/6058))
+- Fixed an issue where neither Python nor created venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))
+- Improved error message of `poetry publish` in the event of an upload error. ([#6043](https://github.com/python-poetry/poetry/pull/6043))
```
Comment (mkniewallner) [inline CHANGELOG.md:22]:
```suggestion
- Improved error message of `poetry publish` in the event of an upload error ([#6043](https://github.com/python-poetry/poetry/pull/6043))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,33 @@
...
+- Fixed a tomlkit regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))
```
Comment (mkniewallner) [inline CHANGELOG.md:23]:
```suggestion
- Fixed a `tomlkit` regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,33 @@
...
+
+
+### Docs
+- Documented the use of the "subdirectory" parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
+- Documented tox config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))
```
Comment (mkniewallner) [inline CHANGELOG.md:28]:
```suggestion
- Documented `tox` config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,33 @@
...
+- Fixed an issue where Windows drive mappings break virtual environment names ([#6110](https://github.com/python-poetry/poetry/pull/6110))
+- Fixed an issue where symlinks in lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))
+- Fixed an issue where `poetry lock` fails without output ([#6058](https://github.com/python-poetry/poetry/pull/6058))
+- Fixed an issue where neither Python nor created venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))
+- Improved error message of `poetry publish` in the event of an upload error. ([#6043](https://github.com/python-poetry/poetry/pull/6043))
+- Fixed a tomlkit regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))
+
+
+### Docs
+- Documented the use of the "subdirectory" parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
```
Comment (mkniewallner) [inline CHANGELOG.md:27]:
```suggestion
- Documented the use of the `subdirectory` parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,33 @@
 # Change Log
 
+## [1.2.0rc1] - 2022-08-16
+
+### Added
+- Virtual environments can now be created even with empty project name ([#5856](https://github.com/python-poetry/poetry/pull/5856))
```
Comment (mkniewallner) [inline CHANGELOG.md:6]: I think this change belongs to `Fixed` (or `Changed`) rather than `Added`?

**Comment by radoering** [comment]
New entry:

- Fixed an infinite loop in the solver ([#6178](https://github.com/python-poetry/poetry/pull/6178))

File: CHANGELOG.md (hunk 9 above)
Comment (finswimmer) [reply CHANGELOG.md]: I wasn't sure about it. IMO an empty project name isn't useful and therefor it was never considered as supported. Now it is, that's why I put it into this section. But if there are strong feelings against this, I can change it.

**Review by mkniewallner — APPROVED** [review]

File: CHANGELOG.md (hunk 9 above)
Comment (mkniewallner) [reply CHANGELOG.md]: No strong feelings, I'm fine with keeping it here.

**Review by neersighted — CHANGES_REQUESTED** [review]
Blocking this on #6185, after that is finished/in, LGTM.

**Comment by mkniewallner** [comment]
Just need to add https://github.com/python-poetry/poetry/pull/6185 (which was a blocker) in `Fixed`, and https://github.com/python-poetry/poetry/pull/5841 in `Added`, and we should be good to go.

**Review by mkniewallner — DISMISSED** [review]

**Review by mkniewallner — DISMISSED** [review]

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,41 @@
 # Change Log
 
+## [1.2.0rc1] - 2022-08-22
+
+### Added
+
+- Added support for subdirectories in git dependencies ([#5172](https://github.com/python-poetry/poetry/pull/5172))
+- Added support for yanked releases and files (PEP-592) ([#5841](https://github.com/python-poetry/poetry/pull/5841))
+- Virtual environments can now be created even with empty project name ([#5856](https://github.com/python-poetry/poetry/pull/5856))
```
Comment (neersighted) [inline CHANGELOG.md:9]:
```suggestion
- Virtual environments can now be created even with empty project names ([#5856](https://github.com/python-poetry/poetry/pull/5856))
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,41 @@
...
+- Added support for `nushell` on `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))
```
Comment (neersighted) [inline CHANGELOG.md:10]:
```suggestion
- Added support for `nushell` in `poetry shell` ([#6063](https://github.com/python-poetry/poetry/pull/6063))
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,41 @@
...
+
+### Changed
+
+- Poetry now fallback to gather metadata for dependencies via pep517 if parsing `pyproject.toml` fail ([#5834](https://github.com/python-poetry/poetry/pull/5834))
```
Comment (neersighted) [inline CHANGELOG.md:14]:
```suggestion
- Poetry now falls back to gather metadata for dependencies via pep517 if parsing `pyproject.toml` fails ([#5834](https://github.com/python-poetry/poetry/pull/5834))
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,41 @@
...
+- Replaced Poetry's helper method `canonicalize_name()` by `packaging.utils.canonicalize_name()` ([#6022](https://github.com/python-poetry/poetry/pull/6022))
```
Comment (neersighted) [inline CHANGELOG.md:15]:
```suggestion
- Replaced Poetry's helper method `canonicalize_name()` with `packaging.utils.canonicalize_name()` ([#6022](https://github.com/python-poetry/poetry/pull/6022))
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,41 @@
...
+- Removed code for the `export` command, which is now provided via plugin ([#6128](https://github.com/python-poetry/poetry/pull/6128))
+- Extras and extras dependencies are now sorted in lock file ([#6169](https://github.com/python-poetry/poetry/pull/6169))
```
Comment (neersighted) [inline CHANGELOG.md:17]:
```suggestion
- Extras and extras dependencies are now sorted in the lock file ([#6169](https://github.com/python-poetry/poetry/pull/6169))
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,41 @@
...
+- Removed deprecated CLI options ([#6210](https://github.com/python-poetry/poetry/pull/6210))
```
Comment (neersighted) [inline CHANGELOG.md:18]:
```suggestion
- Removed deprecated (1.2-only) CLI options ([#6210](https://github.com/python-poetry/poetry/pull/6210))
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,41 @@
...
+
+### Fixed
+
+- Fixed an issue where symlinks in lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))
```
Comment (neersighted) [inline CHANGELOG.md:22]:
```suggestion
- Fixed an issue where symlinks in the lock file were not resolved ([#5850](https://github.com/python-poetry/poetry/pull/5850))
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,41 @@
...
+- Fixed a `tomlkit` regression resulting in inconsistent line endings ([#5870](https://github.com/python-poetry/poetry/pull/5870))
+- Fixed an issue where `POETRY_PYPI_TOKEN_PYPI` environment variable wasn't picked up ([#5911](https://github.com/python-poetry/poetry/pull/5911))
```
Comment (neersighted) [inline CHANGELOG.md:24]:
```suggestion
- Fixed an issue where the `POETRY_PYPI_TOKEN_PYPI` environment variable wasn't respected ([#5911](https://github.com/python-poetry/poetry/pull/5911))
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,5 +1,41 @@
...
+- Fixed an issue where neither Python nor created venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))
```
Comment (neersighted) [inline CHANGELOG.md:25]:
```suggestion
- Fixed an issue where neither Python nor a managed venv can be found, when using Python from MS Store ([#5931](https://github.com/python-poetry/poetry/pull/5931))
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,5 +1,41 @@
...
+- Improved error message of `poetry publish` in the event of an upload error ([#6043](https://github.com/python-poetry/poetry/pull/6043))
+- Fixed an issue where `poetry lock` fails without output ([#6058](https://github.com/python-poetry/poetry/pull/6058))
+- Fixed an issue where Windows drive mappings break virtual environment names ([#6110](https://github.com/python-poetry/poetry/pull/6110))
+- `tomlkit` versions with memory leak are now avoided ([#6160](https://github.com/python-poetry/poetry/pull/6160))
+- Fixed an infinite loop in the solver ([#6178](https://github.com/python-poetry/poetry/pull/6178))
+- Fixed an issue where latest version was used instead of locked one for vcs dependencies with extras ([#6185](https://github.com/python-poetry/poetry/pull/6185))
+
+### Docs
+
+- Documented the use of the `subdirectory` parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
```
Comment (neersighted) [inline CHANGELOG.md:35]:
```suggestion
- Document use of the `subdirectory` parameter ([#5949](https://github.com/python-poetry/poetry/pull/5949))
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,5 +1,41 @@
...
+- Documented `tox` config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))
```
Comment (neersighted) [inline CHANGELOG.md:36]:
```suggestion
- Document suggested `tox` config for different use cases ([#6026](https://github.com/python-poetry/poetry/pull/6026))
```

**Review by neersighted — CHANGES_REQUESTED** [review]

**Review by neersighted — APPROVED** [review]

**Pull request merged**